dsnp-wallet/dsnp-wallet/Source/ViewControllers/ProfileViewController.swift

```diff
@@ -5,10 +5,366 @@
+            }
+        )
+    }
+    
+    @objc private func keyboardWillHide(notification:NSNotification){
+        let duration = notification.userInfo?[UIResponder.keyboardAnimationDurationUserInfoKey] as! Double
+        let curve = notification.userInfo![UIResponder.keyboardAnimationCurveUserInfoKey] as! UInt
+        
+        UIView.animate(
+            withDuration: duration,
+            delay: 0.0,
+            options: UIView.AnimationOptions.init(rawValue: curve),
+            animations: {
+                self.scrollViewBottomConstraint.constant = 0
+                self.saveViewBottomConstraint.constant = CGFloat(-1 * self.saveBtnBottomOffset)
+                self.view.layoutIfNeeded()
+            }
+        )
+    }
+    
+    @objc private func dismissKeyboard() {
+        view.endEditing(true)
+    }
+}
+
+extension ProfileViewController: UITextFieldDelegate {
+    func textFieldDidChangeSelection(_ textField: UITextField) {
+        saveBtn(enabled: isEdited())
+    }
+}
+
+extension ProfileViewController: ProfileHeaderDelegate {
+    func tappedAvatar() {
+        presentActionSheet()
+    }
+}
+
+extension ProfileViewController: UIImagePickerControllerDelegate, UINavigationControllerDelegate {
+    private func presentActionSheet() {
+        self.view.endEditing(true)
+        
+        let actionSheetController = UIAlertController(title: nil, message: nil, preferredStyle: .actionSheet)
+        
+        let takeAction = UIAlertAction(title: "Take photo", style: .default) { _ in
+            let vc = UIImagePickerController()
+            vc.delegate = self
+            vc.sourceType = UIImagePickerController.SourceType.camera
+            vc.allowsEditing = true
+            self.present(vc, animated: true)
+        }
+                
+        let uploadAction = UIAlertAction(title: "Upload photo", style: .default) { _ in
+            let vc = UIImagePickerController()
+            vc.sourceType = .photoLibrary
+            vc.allowsEditing = true
+            vc.delegate = self
+            self.present(vc, animated: true)
+        }
+        
+        let cancelAction = UIAlertAction(title: "Cancel", style: .cancel) { _ in
+            actionSheetController.dismiss(animated: true, completion: nil)
+        }
+        
+        actionSheetController.addAction(takeAction)
+        actionSheetController.addAction(uploadAction)
+        actionSheetController.addAction(cancelAction)
+        
+        self.present(actionSheetController, animated: true, completion: nil)
+    }
+    
+    internal func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [UIImagePickerController.InfoKey : Any]) {
+        guard let image = info[UIImagePickerController.InfoKey.editedImage] as? UIImage,
+              let image = image.downsized(newWidth: 512),
+              let profileHeaderView = profileHeaderView else { return }
+        profileHeaderView.imageView.image = nil
```

Comment:
Are we not currently setting profileHeaderView.imageView to the selected image?

Reply:
Updated to set the image